File: resources/capability-bundle/.claude/skills/lego-extraction/scripts/validate-candidate.py
import json
import sys


candidate = json.loads(sys.argv[1] if len(sys.argv) > 1 else "{}")
required = [
    "semantic_core",
    "logical_skeleton",
    "language_shells",
    "retrieval_cues",
    "scope",
    "provenance",
]

errors = [
    f"missing:{key}"
    for key in required
    if candidate.get(key) is None or candidate.get(key) == ""
]

if not isinstance(candidate.get("language_shells"), list) or not candidate["language_shells"]:
    errors.append("language_shells must be a non-empty array")

if len(str(candidate.get("semantic_core", ""))) > 280:
    errors.append("semantic_core is probably too large for one module")

sys.stdout.write(json.dumps({"valid": not errors, "errors": errors}, separators=(",", ":")))

File: resources/capability-bundle/.claude/skills/lego-extraction/scripts/validate-candidate.mjs
const candidate = JSON.parse(process.argv[2] ?? '{}');
const required = [
  'semantic_core',
  'logical_skeleton',
  'language_shells',
  'retrieval_cues',
  'scope',
  'provenance',
];

const errors = required
  .filter((key) => candidate[key] == null || candidate[key] === '')
  .map((key) => `missing:${key}`);

if (!Array.isArray(candidate.language_shells) || candidate.language_shells.length === 0) {
  errors.push('language_shells must be a non-empty array');
}

if (String(candidate.semantic_core ?? '').length > 280) {
  errors.push('semantic_core is probably too large for one module');
}

process.stdout.write(JSON.stringify({ valid: errors.length === 0, errors }));
